### PR TITLE
Api docs for ResponseCaching

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Part of #26397.

Breaking it up into smaller PRs for review. This one covers ResponseCaching, StaticFiles, HostFilter and HttpsPolicy.

### CODE REVIEW

File: src/Middleware/ResponseCaching/src/ResponseCachingFeature.cs [hunk 1]
```diff
@@ -5,10 +5,12 @@
 
 namespace Microsoft.AspNetCore.ResponseCaching
 {
+    /// <inheritdoc />
```
Comment (pranavkm) [inline src/Middleware/ResponseCaching/src/ResponseCachingFeature.cs:8]:
```suggestion
    /// Default implementation for <see cref="IResponseCachingFeature" />
```

The methods are fine, but it's odd to have the same summary as the contract.

File: src/Middleware/ResponseCaching/src/ResponseCachingOptions.cs [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System.ComponentModel;
 
 namespace Microsoft.AspNetCore.ResponseCaching
 {
+    /// <summary>
+    /// Options for configuring the <see cref="ResponseCachingMiddleware"/>.
+    /// </summary>
     public class ResponseCachingOptions
```
Comment (pranavkm) [inline src/Middleware/ResponseCaching/src/ResponseCachingOptions.cs:11]: * For the `SizeLimit` property below, do you think it's valuable to explain what happens once the limit is hit?
* For `MaximumBodySize` should we add what happens if the limit is exceeded?

**Review by Tratcher — APPROVED** [review]

File: src/Middleware/ResponseCaching/src/ResponseCachingOptions.cs (hunk 2 above)
Comment (JunTaoLuo) [reply src/Middleware/ResponseCaching/src/ResponseCachingOptions.cs:11]: Updated

**Comment by JunTaoLuo** [comment]
@pranavkm :up::date:

**Review by pranavkm — APPROVED** [review]

**Pull request merged**